src/lcm/discrete_problem.py

```diff
@@ -0,0 +1,280 @@
+"""Functions that aggregate the conditional continuation values over discrete choices.
+
+By conditional continuation value we mean continuation values conditional on a discrete
+choice, i.e. the result of solving the continuous choice problem conditional on the
+discrete choice. These are also _conditional_ on a given state.
+
+By aggregate we mean calculating the expected maximum of the continuation values, based
+on the distribution of utility shocks. Currently we support no shocks. In the future,
+we will at least support IID Extreme Value Type 1 shocks.
+
+How we (want to) solve the problem:
+-----------------------------------
+
+- No shocks: We take the maximum of the conditional continuation values.
+
+- IID Extreme Value Type 1 shocks: We do a logsum calculation.
+
+"""
+
+from collections.abc import Callable
+from functools import partial
+from typing import Any, Literal
+
+import jax
+import jax.numpy as jnp
+import pandas as pd
+from jax import Array
+
+from lcm.typing import SegmentInfo
+
+
+def get_solve_discrete_problem(
+    *,
+    shock_type: Literal["extreme_value"] | None,
+    variable_info: pd.DataFrame,
+    is_last_period: bool,
+    choice_segments: SegmentInfo | None,
+    params: dict[str, Any],
+) -> Callable[[Array], Array]:
+    """Get function that computes the expected max. of conditional continuation values.
+
+    The maximum is taken over the discrete and sparse choice variables in each state.
+
+    Args:
+        shock_type (Literal["extreme_value"] | None): Type of choice shock. Currently
+            only None is supported. Work for "extreme_value" is in progress.
+        variable_info (pd.DataFrame): DataFrame with information about the variables.
+        is_last_period (bool): Whether the function is created for the last period.
+        choice_segments (SegmentInfo): Dictionary with the entries "segment_ids" and
```

Comment:
I think more important than the precise entries is the description of the content: This is how sparse choices are stored, right?